### PR TITLE
CEM-2350 - create a nutrition fact label component

### PR DESCRIPTION
**Pull request by roggc**
nutritionfact

### CODE REVIEW

**Comment by ralph-dev** [comment]
Send me the video where you can edit the values. Are the default values 0 upon creating

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+export const NutritionFact:React.FC<INutritionFactProps>=({entries}):React.ReactElement=>
+{
+
+    /**
+     * renders an entry of type entry
+     * @param entry {IEntryProps} - the entry to render
+     * @returns {React.ReactElement} the rendered entry
+     */
+    const renderEntry=useCallback((entry:IEntryProps)=>{
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:29]: Return typing missing for functions

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    unity,
+    fontSize=smallFontSize,
+    bold=false,
+    separatorWidth=1,
+    indentationNumber=0,
+    indentationSize=4,
+    ...props
+}):React.ReactElement=>{
+
+    const dailyValue= useMemo(()=>{
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:123]: dailyValuePercentage where is JSDOC?

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+    /**
+     * computes the amount of white space
+     * @returns {string} white space 
+     */
+    const renderIndentation=useCallback(()=>{
+        let space=''
+        const nbsp='\u00a0'
+        for(let j=0;j<indentationNumber;j+=1){
+            for(let i=0;i<indentationSize; i+=1) space+=nbsp
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:137]: Use bracket for for loop

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 4]
```diff
@@ -0,0 +1,196 @@
+import React,{useCallback,useMemo} from 'react'
+import styled from 'styled-components'
+import { Mixins } from '../../Utils'
+import {
+    Main,
+    MainInterface,
+} from '../../Utils/BaseStyles';
+import Theme from '../../Themes/ThemeTemplate'
+
+const smallFontSize=Theme.font.size.small
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:10]: Why is this not in styled components?

**Review by ralph-dev — CHANGES_REQUESTED** [review]
Code quality

**Comment by ralph-dev** [comment]
Fix the styling of the input 😂

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+import Theme from '../../Themes/ThemeTemplate'
+import {Input} from '../../Inputs/Input/Input'
+
+const EXTRA_PIXEL=1
+const ENTRY_INPUT_WIDTH=20
+const HEADER_ENTRY_INPUT_WIDTH=40
+
+interface IAdditionalProps{
+    ref?:boolean;
+    delay?:boolean;
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:17]: What is this used for?

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+}
+
+interface IEntries{
+    type:string;
+    entry:IEntryProps&IAdditionalProps|IHeadingEntryProps&IAdditionalProps;
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:22]: Try to use TypeScript Comments
Interface {
  /* Data Per Line */ <--- start writing this
  Entry: Type
}

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+    const infoRef=useRef<IInfo|null>(null)
+
+    /**
+     * renders an entry of type entry
+     * @param entry {IEntryProps} - the entry to render
+     * @returns {React.ReactElement} the rendered entry
+     */
+    const renderEntry=useCallback((entry:IEntryProps&IAdditionalProps):React.ReactElement=>{
+        const {label,...rest}=entry
+        return <Entry key={label} label={label} {...rest} margin='0 0 2px' padding='0 0 1px' editMode={editMode} />
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:47]: Won't the padding always be overrided by the padding. Should ...rest be at the end :)

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 8]
```diff
@@ -0,0 +1,296 @@
...
+    /**
+     * updates state for second label based on event fired
+     * @param event {React.ChangeEvent<HTMLInputElement>} - the fired event 
+     */
+    const updateSecondLabel=useCallback(({target}:React.ChangeEvent<HTMLInputElement>)=>{
+        setSecondLabelState(target.value)
+    },[]) 
+
+    /**
+     *
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:150]: Missing doc

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 9]
```diff
@@ -0,0 +1,296 @@
...
+    const dailyValuePercentage= useMemo(()=>{
+        if(dailyAmount){
+            /**
+             * gets the percentage of a value respect a totalValue
+             * @param value {number} - value for which we want to get percentage
+             * @param totalValue {number} - value for which percentage is 100 
+             * @returns {number} the percentage
+             */
+            const getPercentage=(value:number,totalValue:number)=>Math.round(value/totalValue*100)
+            return getPercentage(parseInt(amountState,10),dailyAmount)
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:205]: What and why?

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 10]
```diff
@@ -0,0 +1,296 @@
...
+                <Bold bold={bold}>
+                    {label}
+                    &nbsp;
+                </Bold>
+                <AmountContainer>
+                    {renderAmount()}
+                    {unity}
+                </AmountContainer>
+            </LabelContainer>
+            {dailyValuePercentage!==null&&(
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:252]: Cant you use
!dailyValuePercentage instead? What is the value is undefined...

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 11]
```diff
@@ -0,0 +1,296 @@
...
+
+interface IEntryContainerProps extends MainInterface{
+    fontSize?:string;
+    separatorWidth:number;
+    justifyContent?:string;
+    bold?:boolean;
+}
+
+const EntryContainer=styled.div<IEntryContainerProps>`
+${({separatorWidth,fontSize=Theme.font.size.small,justifyContent='flex-start',bold=false,...props}):string=>`
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:270]: One per line please

File: src/Fragments/InputFragment.tsx [hunk 12]
```diff
@@ -46,6 +47,20 @@ const InputElement = styled.input<InputFragmentProps>`
...
+                return `width:${width}px;`
+            case 'string':
+                return `width:${width};`
+            default:
+                return ''
+            }
+        }
+        return ''
+    }}
+
```
Comment (ralph-dev) [inline src/Fragments/InputFragment.tsx:63]: Why is this needed?

**Review by ralph-dev — CHANGES_REQUESTED** [review]
Code Quality. don't rush!

File: src/Containers/NutritionFact/NutritionFact.tsx (hunk 7 above)
Comment (roggc) [reply src/Containers/NutritionFact/NutritionFact.tsx:47]: entry is of type IEntryProps&IAdditionalProps which has no padding in it, so {...rest} contain no padding and will not be overwrited. In this case I this case I think it is not necessary {...rest} to be at the end 😉

File: src/Containers/NutritionFact/NutritionFact.tsx (hunk 10 above)
Comment (roggc) [reply src/Containers/NutritionFact/NutritionFact.tsx:252]: !dailyValuePercentage don't work because we want also to represent (display) values 0. dailyValuePercentage could not be undefined because it's of type number|null

File: src/Fragments/InputFragment.tsx (hunk 12 above)
Comment (roggc) [reply src/Fragments/InputFragment.tsx:63]: because if I do not this Input was not responding to setting of width prop

<Input ... width={65} /> or <Input ... width='65px' /> did not work if I do not do this.

File: src/Containers/NutritionFact/NutritionFact.tsx (hunk 9 above)
Comment (roggc) [reply src/Containers/NutritionFact/NutritionFact.tsx:205]: this is because in some place of edition I was using it in more than one place so I extracted logic of getting percentage into a function. later on I finally ended up only using in one place.

File: src/Containers/NutritionFact/NutritionFact.tsx (hunk 5 above)
Comment (roggc) [reply src/Containers/NutritionFact/NutritionFact.tsx:17]: this is to set width of the component dynamically. if I do not this then width of the component gets like this:
Captura de pantalla 2021-03-28 a las 9 23 39
because of the last entry of daily value explication. so to fix that we use delay:true on this last entry and ref:true in the first heading entry, which is the one we want to take width from to set it in the main container as max-width, because in this case the first heading is the one which takes more width

Comment (ralph-dev) [reply src/Containers/NutritionFact/NutritionFact.tsx:19]: Why is it named delay??
Can you put comments directly on the code

File: src/Containers/NutritionFact/NutritionFact.tsx (hunk 9 above)
Comment (ralph-dev) [reply src/Containers/NutritionFact/NutritionFact.tsx:210]: Then fix it, don't leave it like this if your not reusing it

File: src/Containers/NutritionFact/NutritionFact.tsx (hunk 10 above)
Comment (ralph-dev) [reply src/Containers/NutritionFact/NutritionFact.tsx:257]: Then use ?? Instead.

File: src/Fragments/InputFragment.tsx (hunk 12 above)
Comment (ralph-dev) [reply src/Fragments/InputFragment.tsx:63]: What do you mean, the width can be modified before no?

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 13]
```diff
@@ -0,0 +1,307 @@
...
+    editMode?:boolean;
+}
+
+export const NutritionFact:React.FC<INutritionFactProps>=({entries,editMode}):React.ReactElement=>
+{
+    const mainContainerRef=useRef<HTMLDivElement>(null)
+
+    const containerRef=useRef<HTMLDivElement>(null)
+
+    const delayLabel=useRef<string[]>([])
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:40]: Use better names

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 14]
```diff
@@ -0,0 +1,307 @@
...
+export interface INutritionFactProps{
+    entries:IEntries[];
+    editMode?:boolean;
+}
+
+export const NutritionFact:React.FC<INutritionFactProps>=({entries,editMode}):React.ReactElement=>
+{
+    const mainContainerRef=useRef<HTMLDivElement>(null)
+
+    const containerRef=useRef<HTMLDivElement>(null)
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:38]: What does this contain versus the main container?

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 15]
```diff
@@ -0,0 +1,307 @@
...
+}
+
+export interface INutritionFactProps{
+    entries:IEntries[];
+    editMode?:boolean;
+}
+
+export const NutritionFact:React.FC<INutritionFactProps>=({entries,editMode}):React.ReactElement=>
+{
+    const mainContainerRef=useRef<HTMLDivElement>(null)
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:36]: rootContainer?

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 16]
```diff
@@ -0,0 +1,307 @@
...
+            return <HeadingEntry key={label} label='' infoRef={infoRef} editMode={editMode} {...rest} />
+        }
+        return <HeadingEntry key={label} label={label} editMode={editMode} {...rest} />
+    }
+    ,[])
+
+    /**
+     * this sets max width for the main container of the component and also sets the content for the delayed label
+     */
+    useEffect(()=>{
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:73]: Why not useLayoutEffect?

**Review by ralph-dev — CHANGES_REQUESTED** [review]
Code Quality

File: src/Containers/NutritionFact/NutritionFact.tsx (hunk 16 above)
Comment (roggc) [reply src/Containers/NutritionFact/NutritionFact.tsx:73]: I tried useLayoutEffect but then the content of the heading for daily value explanation doesn't appear.

File: src/Containers/NutritionFact/NutritionFact.tsx (hunk 14 above)
Comment (roggc) [reply src/Containers/NutritionFact/NutritionFact.tsx:38]: this is a ref to the entry of type 'heading' from which we will take its width to set it as max-width for the rootContainer (mainContainer here)

File: src/Containers/NutritionFact/NutritionFact.tsx (hunk 10 above)
Comment (roggc) [reply src/Containers/NutritionFact/NutritionFact.tsx:257]: no, ?? doesn't serve because someVar??expresion executes expresion if someVar is null, and I don't want this, I want the contrary

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 17]
```diff
@@ -71,7 +71,7 @@ export const NutritionFact:React.FC<INutritionFactProps>=({entries,editMode}):Re
      * this sets max width for the main container of the component and also sets the content for the delayed label
      */
     useLayoutEffect(()=>{
-        if(rootContainerRef.current&&containerRef.current) rootContainerRef.current.style.maxWidth=`${containerRef.current.clientWidth+EXTRA_PIXEL}px`
+        if(rootContainerRef.current&&headingContainerRef.current) rootContainerRef.current.style.maxWidth=`${headingContainerRef.current.clientWidth+EXTRA_PIXEL}px`
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:74]: Why do you need this XD Why not let the width be dynamic and flex correctly?

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 18]
```diff
@@ -0,0 +1,299 @@
...
+    ...props
+}):React.ReactElement=>{
+    const [amountState,setAmountState]=useState<string>(amount.toString())
+    
+    /**
+     * this computes the daily value percentage
+     */
+    const dailyValuePercentage= useMemo(()=>{
+        if(dailyAmount)
+            return Math.round(parseInt(amountState,10)/dailyAmount*100)
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:203]: 10 and 100 are magic numbers

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 19]
```diff
@@ -0,0 +1,299 @@
...
+    indentationNumber=0,
+    indentationSize=4,
+    ...props
+}):React.ReactElement=>{
+    const [amountState,setAmountState]=useState<string>(amount.toString())
+    
+    /**
+     * this computes the daily value percentage
+     */
+    const dailyValuePercentage= useMemo(()=>{
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:201]: missing TypeScript Return types for function

const Example Function  = (): void => {}

**Review by ralph-dev — COMMENTED** [review]
Code Quality

File: src/Containers/NutritionFact/NutritionFact.tsx (hunk 17 above)
Comment (roggc) [reply src/Containers/NutritionFact/NutritionFact.tsx:74]: I told you before. If I do not do this then the width of the component its the width of its largest one, which is when I explain what a daily value is at the bottom, and we do not want this

File: src/Containers/NutritionFact/NutritionFact.tsx (hunk 19 above)
Comment (roggc) [reply src/Containers/NutritionFact/NutritionFact.tsx:201]: first of all, it's not a function, it's a value. and second, its typed is implicit. 
Captura de pantalla 2021-03-30 a las 13 56 36

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 20]
```diff
@@ -0,0 +1,301 @@
...
+const EXTRA_PIXEL=1
+const ENTRY_INPUT_WIDTH=20
+const HEADER_ENTRY_INPUT_WIDTH=40
+const DECIMAL_BASIS=10
+const PERCENTAGE_FACTOR=100
+
+interface IAdditionalProps{
+    /* if the heading entry must to be taken as the reference entry for taking width from */
+    ref?:boolean;
+    /* if the content of the heading entry must be delayed after setting max width of the main container */
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:20]: IgnoreWidthOfContent
- Ignoring the Width of the Entry so that it doesn't affect the Max Width of the Entire Component.

Comment (ralph-dev) [reply src/Containers/NutritionFact/NutritionFact.tsx:20]: Booleans always start with is.
const getCartTotal - What the End Result Is.
const getSubTotalOfItems+TaxReturnTotal - Exactly what the code is doing.

isIgnoringWidthOfContent

Comment (ralph-dev) [reply src/Containers/NutritionFact/NutritionFact.tsx:20]: isWidthIgnored

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 21]
```diff
@@ -0,0 +1,301 @@
...
+
+const EXTRA_PIXEL=1
+const ENTRY_INPUT_WIDTH=20
+const HEADER_ENTRY_INPUT_WIDTH=40
+const DECIMAL_BASIS=10
+const PERCENTAGE_FACTOR=100
+
+interface IAdditionalProps{
+    /* if the heading entry must to be taken as the reference entry for taking width from */
+    ref?:boolean;
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:19]: Pick One Entry to Specify as the Max Width of the Component.
isMaxWidthAdjustedtoThisComponent

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 22]
```diff
@@ -0,0 +1,277 @@
...
+
+const EXTRA_PIXEL=1
+const ENTRY_INPUT_WIDTH=20
+const HEADER_ENTRY_INPUT_WIDTH=40
+const DECIMAL_BASIS=10
+const PERCENTAGE_FACTOR=100
+
+interface IAdditionalProps{
+    /* if the heading entry must to be taken as the reference entry for taking width from */
+    isRefForComputingMaxWidth?:boolean;
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:19]: isRefForComputingMaxWidth -> isRefForComputingMaxWidthOfRootContainer
isRefForComputingMaxWidthOfRootContainer -> isMaxWidthOfRootContainer

Comment (ralph-dev) [reply src/Containers/NutritionFact/NutritionFact.tsx:19]: Don't use ref if it's not a ref

Comment (ralph-dev) [reply src/Containers/NutritionFact/NutritionFact.tsx:19]: isMaxWidthOfRootContainer

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 23]
```diff
@@ -0,0 +1,277 @@
...
+}
+
+interface IEntries{
+    /* type of entry, heading or entry */
+    type:string;
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:24]: ENUM of Heading or Entry

Comment (ralph-dev) [reply src/Containers/NutritionFact/NutritionFact.tsx:24]: Type the Enum document the two difference WHATS AN ENTRY?

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 24]
```diff
@@ -0,0 +1,277 @@
...
+    /* data per line */
+    entry:IEntryProps|IHeadingEntryProps&IAdditionalProps;
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:26]: data not entry?

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 25]
```diff
@@ -0,0 +1,277 @@
...
+}
+
+export interface INutritionFactProps{
+    entries:IEntries[];
+    editMode?:boolean;
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:31]: isEditing

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 26]
```diff
@@ -0,0 +1,277 @@
...
+}
+
+export const NutritionFact:React.FC<INutritionFactProps>=({entries,editMode}):React.ReactElement=>
+{
+    const rootContainerRef=useRef<HTMLDivElement>(null)
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:36]: Comment what this ref is used for

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 27]
```diff
@@ -0,0 +1,277 @@
...
+    },[editMode])
+
+    /**
+     * renders an entry of type heading
+     * @param entry {IHeadingEntryProps} - the entry to render
+     * @returns {React.ReactElement} the rendered entry
+     */
+    const renderHeadingEntry=useCallback((entry:IHeadingEntryProps&IAdditionalProps):React.ReactElement=>{
+        const {label,isRefForComputingMaxWidth,...rest}=entry
+        if(isRefForComputingMaxWidth) return <HeadingEntry key={label} label={label} ref={headingLabelRef} editMode={editMode} {...rest} />
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:57]: if () {
code
}

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 28]
```diff
@@ -0,0 +1,277 @@
...
+        return <HeadingEntry key={label} label={label} editMode={editMode} {...rest} />
+    }
+    ,[])
+
+    /**
+     * this sets max width for the root container
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:63]: based on the entry to Compute the width

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 29]
```diff
@@ -0,0 +1,277 @@
...
+        if(rootContainerRef.current&&headingLabelRef.current) rootContainerRef.current.style.maxWidth=`${headingLabelRef.current.clientWidth+EXTRA_PIXEL}px`
+    },[])
+
+    /**
+     * renders the entries
+     */
+    const renderEntries=useCallback(()=>
+        entries.map(({type,entry})=>{
+            switch(type){
+            case 'entry':
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:75]: ENUMs

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 30]
```diff
@@ -0,0 +1,277 @@
...
+            case 'heading':
+                return renderHeadingEntry(entry as IHeadingEntryProps&IAdditionalProps)
+            default:
+                return null
+            }
+        })
+    ,[entries])
+
+    return     (
+        <RootContainer padding='5px' ref={rootContainerRef}>
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:86]: move padding into the Styled Component

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 31]
```diff
@@ -0,0 +1,277 @@
...
+border:2px solid black;
+${({width,...props}):string=>`
+${width?`width:${width}px;`:'width:fit-content;'}
+${Main({...props})}
+`}
+`
+
+interface ICommonEntryProps{
+    fontSize?:string;
+    bold?:boolean;
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:106]: isBold

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 32]
```diff
@@ -0,0 +1,277 @@
...
+    separatorWidth?:number;
+    editMode?:boolean;
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:108]: isEditing

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 33]
```diff
@@ -0,0 +1,277 @@
...
+                return <Input value={secondLabelState} onChange={updateSecondLabel} width={HEADER_ENTRY_INPUT_WIDTH} />
+            return <div>{secondLabelState}</div>
+        }
+        return null
+    },[secondLabelState])
+
+    return (
+        <EntryContainer separatorWidth={separatorWidth} {...props}>
+            <div ref={ref}>{label}</div>
+            {renderSecondLabel()}
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:148]: LeftText
rightText

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 34]
```diff
@@ -0,0 +1,277 @@
...
+        const nbsp='\u00a0'
+        for(let j=0;j<indentationNumber;j+=1){
+            for(let i=0;i<indentationSize; i+=1) {
+                space+=nbsp
+            }
+        }
+        return space
+    },[indentationNumber,indentationSize])
+
+    const updateAmountState=({target}:React.ChangeEvent<HTMLInputElement>)=>{
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:200]: Amount vendor inputted

File: src/Containers/NutritionFact/NutritionFact.tsx [hunk 35]
```diff
@@ -0,0 +1,277 @@
...
+            )}
+        </EntryContainer>
+    )
+}
+
+interface IEntryContainerProps extends MainInterface{
+    fontSize?:string;
+    separatorWidth:number;
+    justifyContent?:string;
+    bold?:boolean;
```
Comment (ralph-dev) [inline src/Containers/NutritionFact/NutritionFact.tsx:241]: isBold

**Review by ralph-dev — APPROVED** [review]
LGTM, Fix the Comments, Merge the PR
